report_xlsx/report/report_abstract_xlsx.py

```diff
@@ -10,6 +11,56 @@
 
 try:
     import xlsxwriter
+
+    class PatchedXlsxWorkbook(xlsxwriter.Workbook):
+        def _check_sheetname(self, sheetname, is_chartsheet=False):
+            """We want to avoid duplicated sheet names exceptions the same following
+            the same philosophy that Odoo implements overriding the main library
+            to avoid the 31 characters limit triming the strings before sending them
+            to the library.
+
+            In some cases, there's not much control over this as the reports send
+            automated data and the potential exception is hidden underneath making it
+            hard to debug the original issue. Even so, different names can become the
+            same one as their strings are trimmed to those 31 character limit.
+
+            This way, once we come across with a duplicated, we set that final 3
+            characters with a sequence that we evaluate on the fly. So for instance:
+
+            - 'Sheet name' will be 'Sheet name~01'
```

Comment:
Just wondering, with this change, the sheet will be named always this way or only when you choose it?

In the first case, that might be complicated, as you are changing the current output of the document.